docs/source/hsm.md

```diff
@@ -89,31 +108,31 @@ in the `signcerts` folder to retrieve the private key from inside the HSM.
 The process for creating the MSP folders will be different depending on if you
 are using a Fabric Certificate Authority (CA) your own CA.
 
-### Using a Fabric CA
+### Using an HSM with a Fabric CA
+
+You can set up a Fabric CA to use an HSM by making the same edits to the CA server configuration file as you would make to a peer or ordering node. Because you can use the Fabric CA to generate keys inside an HSM, the process of creating the local MSP folders is straightforward. Use the following steps:
+
+1. Create an HSM token and point to it in the Fabric CA server configuration file. When the Fabric CA server starts, the HSM described in the configuration file will generate the CA signing certificate and store the private key inside the HSM. If you are not concerned about exposing your CA signing certificate, you can skip this step and only configure an HSM for your peer or ordering nodes.
 
-You can set up a Fabric CA to use an HSM by making the same edits to the
-configuration file as you would make to a peer or ordering node. Because you can
-use Fabric CA to generate keys inside an HSM, the process of creating the local
-MSP folders is straightforward. Use the following steps:
+2. Use the Fabric CA client to register the peer or ordering node identities with your CA.
 
-1. Create an HSM token and point to it in the Fabric CA server configuration
-file. When the Fabric CA server starts, it will generate the CA signing
-certificate inside your HSM. If you are not concerned about exposing your CA
-signing certificate, you can skip this step.
+3. Before you deploy a peer or ordering node with HSM, you need to enroll the node admin, storing its private key in the HSM. Edit the BCCSP section of the Fabric CA client config file or associated environment variables to point to the HSM configuration for your peer or ordering node. In the Fabric CA Client configuration file, replace the default `SW` configuration with the `PKCS11` configuration and providing the values for your own HSM:
 
-2. Use the Fabric CA client to register the peer or ordering node identities
-with your CA.
+  ```
+  bccsp:
+    default: PKCS11
+    pkcs11:
+      Library: /etc/hyperledger/fabric/libsofthsm2.so
+      Pin: 71811222
+      Label: fabric
+      hash: SHA2
+      security: 256
+      Immutable: false
```

Comment:
Is there any reason why some of these are capitalized while others aren't?

Reply:
You would have to ask the developers that.

Reply:
There is no specific reason. What I can think of is , Capital letter characterizes HSM related attributes while lower case are generic security attributes.

Reply:
I talked to @jyellick about this and it appears the standard we're trying to move towards is lowerCamelCase. Unfortunately, I'm not too sure if simply changing all the fields to lower case will still work as intended with how fabric currently loads the yaml. For now it's probably fine as is until we can put together a proper test for verifying the pkcs11 fields are loaded and used properly.

Reply:
In addition, it seems in orderer.yaml the fields are all capitalized, whereas in peer's core.yaml it isnt. Not sure about fabric ca I'll have to verify